utils/github_reporting.py

```diff
@@ -0,0 +1,531 @@
+
+        return pr_list
+
+    def get_update_issues(self):
+        json_data = requests.get(self.get_job_api_url(details=True)).json()
+
+        logs_to_check = []
+
+        for log in json_data['job']['ulogs']:
+            if log.endswith("packages.txt"):
+                logs_to_check.append("{}/tests/{}/file/{}".format(
+                    OPENQA_URL, self.job_id, log))
+
+        packages = set()
+
+        for log_url in logs_to_check:
+            log = requests.get(log_url).text.split('\n')
+            for line in log:
+                package = PackageName(line)
+                if package.package_name:
+                    packages.add(package)
+
+        # Validate if there are no copies of the same package with different
+        # versions
+        for package1 in packages:
+            for package2 in packages:
+                if package1.package_name == package2.package_name and \
+                        package1.version != package2.version:
+                    print(
+                        "Warning: found package {} in two different versions: "
+                        "{} and {}".format(
+                            package1.package_name, package1.version,
+                            package2.version))
+
+        repo = GitHubRepo("updates-status")
+        issue_urls = []
+
+        for p in packages:
+            issue_name = "{} v{} (r{})".format(
+                p.package_name, p.version,
+                json_data['job']['settings']['VERSION'])
+            url = repo.get_issues_by_name(issue_name)
+            if url:
+                issue_urls.append(url)
+
+        return issue_urls
+
+    @staticmethod
+    def get_labels_from_results(results):
+        number_of_failures = sum(len(val) for val in results.values())
+        if number_of_failures:
+            return ['openqa-ok']
+        return ['openqa-failed']
+
+    def format_results(self, results):
+        output_string = "{}\n" \
+                        "Complete test suite and dependencies: {}\n" \
+                        "## Failed tests\n".format(COMMENT_TITLE,
+                                                   self.get_dependency_url())
+        number_of_failures = 0
+
+        for k in results:
+            if results[k]:
+                output_string += '* ' + str(k) + "\n"
+                for fail in results[k]:
+                    output_string += '  * ' + str(fail) + '\n'
+                    number_of_failures += 1
+
+        if not number_of_failures:
+            output_string += "No failures!\n"
+
+        return output_string
+
+    def __str__(self):
+        return self.job_name
+
+
+class GitHubRepo:
+    def __init__(self, repo_name, owner="QubesOS"):
+        self.repo = repo_name
+        self.url = "{}/{}/{}/". format(GITHUB_API_PREFIX, owner, self.repo)
+
+    def get_issues_by_name(self, name):
+        url = self.url + 'issues'
+        while url:
+            data = requests.get(url, headers=github_auth)
```

Comment:
This downloads full list of issues - the same for every call. Please cache it. Maybe a list of `data` objects, so if initial list doesn't have requested issue, the function can download more on demand?